### ISSUE TITLE
Similar project:  gcc analyzer integration tests

### ISSUE THREAD

**Issue by davidmalcolm** [aufover/aufover-benchmark#42]
I just discovered this project via the link in https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108598 and thought you might be interested in a similar project I've been working on:
  https://github.com/davidmalcolm/gcc-analyzer-integration-tests

There are a lot of similarities, and some differences.

My project focuses on just GCC -fanalyzer, and on testing patched versions of GCC; I'm building various upstream C projects with default configuration/build flags, etc, harvesting the results in SARIF form.  I'm not yet bothering with "mock".  I'm also manually classifying the results as true vs false positives, so I can see if one of my patches to GCC improves the output (see the "known-issues" subdirectory).  I have an automated classifier for the Juliet test suite, since that already has a human-edited set of true/false classifications embedded within it.

Are you OK if I copy some of your small testcases into GCC's upstream test suite?  Presumably it's all MIT-licensed?

**Comment by lzaoral** [comment]
Yeah, using our tests is completely fine! The licences are the following:

* `tests/single-c/mem-*` - everything is MIT
* `tests/single-c/predator-*` - everything is GPLv3
    - These are just cherry picks from the regression test suite of Predator taken from [here](https://github.com/kdudka/predator/tree/master/tests/predator-regre). However, the majority of them were written solely by @kdudka and he has no problems with relicensing them under a more permissive license.
* `tests/single-c/arrays-and-loops` - @vmihalko can you please specify the details for these? Thank you!

**Comment by vmihalko** [comment]
* `tests/single-c/arrays-and-loops` - These test are modified cherry picks from [sv-benchamarks](https://gitlab.com/sosy-lab/benchmarking/sv-benchmarks) everything is Apache 2.0.
